core/src/main/java/com/datastax/oss/driver/internal/core/channel/ChannelFactory.java

```diff
@@ -84,8 +88,18 @@ public void setProtocolVersion(ProtocolVersion newVersion) {
     this.protocolVersion = newVersion;
   }
 
-  public CompletionStage<DriverChannel> connect(
+  @VisibleForTesting
+  CompletionStage<DriverChannel> connect(
       final SocketAddress address, DriverChannelOptions options) {
+    return connect(null, address, options);
+  }
+
+  public CompletionStage<DriverChannel> connect(final Node node, DriverChannelOptions options) {
```

Comment:
Needed a way to get at the `NodeMetricUpdater` instance,  couldn't find a means other than passing in the `Node`.  I figured this was ok since outside of unit tests we have access to the `Node` instance, but if there is a better way of getting at the `NodeMetricUpdater` instance might be nice.   This change triggered a lot of unit test code changes, which is why there are a bunch of changes in this PR that don't look related.